src/libraries/Microsoft.PowerFx.LanguageServerProtocol/JsonRpc/FormulaTypeJsonConverter.cs

```diff
@@ -0,0 +1,139 @@
+﻿
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.PowerFx.Core.Public.Types;
+
+namespace Microsoft.PowerFx.LanguageServerProtocol
+{
+    public class FormulaTypeJsonConverter : JsonConverter<FormulaType>

```

Comment:
>FormulaTypeJsonConverter : JsonConverter<FormulaType> [](http://example.com/codeflow?start=17&length=53)

This is DType -->Json ... 
@lesaltzm - do we have similar code already to do this that we can leverage?

Reply:
we need to be really clear on versioning expectations here. 
This is probably something that should live in Fx.Core.

Reply:
Right, yeah. I'm thinking maybe the core of this should be a virtual ToJson function on FormulaType, so it stays up to date if new formula types are added